### PR TITLE
Replace construction of FileInputStream and FileOutputStream objects with Files NIO APIs.

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: src/main/java/org/apache/commons/release/plugin/mojos/CommonsSiteCompressionMojo.java [hunk 1]
```diff
@@ -147,8 +148,9 @@ private void getAllSiteFiles(final File siteDirectory, final List<File> filesToC
      */
     private void writeZipFile(final File outputDirectory, final File directoryToZip, final List<File> fileList)
             throws IOException {
-        try (FileOutputStream fos = new FileOutputStream(outputDirectory.getAbsolutePath() + "/site.zip");
-                ZipOutputStream zos = new ZipOutputStream(fos)) {
+        try (OutputStream fos = Files.newOutputStream(new File(outputDirectory.getAbsolutePath() + "/site.zip")
+                .toPath());
+             ZipOutputStream zos = new ZipOutputStream(fos)) {
```
Comment (kinow) [inline src/main/java/org/apache/commons/release/plugin/mojos/CommonsSiteCompressionMojo.java:153]: Nit-picking, but I think there's one extra space before `ZipOutputStream`, or at least looking at the GitHub interface it appears to have an extra one.

**Review by kinow — COMMENTED** [review]
I have no preference over io and nio here. I normally use io most of the time, except when I need to do some low level operation with input/output streams, or some non-blocking operation. Here I suspect the only difference is that these methods work with `Path`? Or is there another advantage in using `Files.newInputStream`?

The other issue with finalizers, GC, etc, appears to have been fixed in newer versions of Java (this is reported by Sonar/static analyzers) - https://stackoverflow.com/a/55370386

File: src/main/java/org/apache/commons/release/plugin/mojos/CommonsSiteCompressionMojo.java (hunk 1 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/release/plugin/mojos/CommonsSiteCompressionMojo.java:153]: Hi @kinow 
I have check the code directly and i can't see the extra space. I also run the checkstyle and nothing :-(
Regards,

**Comment by arturobernalg** [comment]
> I have no preference over io and nio here. I normally use io most of the time, except when I need to do some low level operation with input/output streams, or some non-blocking operation. Here I suspect the only difference is that these methods work with `Path`? Or is there another advantage in using `Files.newInputStream`?
> 
> The other issue with finalizers, GC, etc, appears to have been fixed in newer versions of Java (this is reported by Sonar/static analyzers) - https://stackoverflow.com/a/55370386

Hi @kinow 
I'm assuming there are users with java 8 version. Maybe in some point this change would not be sense, but i think it's important at the moment
TY

**Pull request merged**